packages/quantic/force-app/main/default/lwc/quanticNotifications/__tests__/quanticNotifications.test.js

```diff
@@ -0,0 +1,189 @@
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticNotifications) {
+      element.setInitializationError();
+    }
+  };
+}
+
+function cleanup() {
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+  jest.clearAllMocks();
+  isInitialized = false;
+}
+
+describe('c-quantic-notifications', () => {
+  beforeAll(() => {
+    mockSuccessfulHeadlessInitialization();
+  });
+
+  afterEach(() => {
+    cleanup();
+    notificationsState = {
+      notifications: exampleNotifications,
+    };
+  });
+
+  describe('when an error occurs during initialization', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    afterAll(() => {
+      mockSuccessfulHeadlessInitialization();
+    });
+
+    it('should display the initialization error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized successfully', () => {
+    describe('when some notifications are fired by the pipeline trigger', () => {
+      it('should render the notifications component', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const notifications = element.shadowRoot.querySelectorAll(
+          selectors.notifications
+        );
+
+        expect(notifications).not.toBeNull();
+        expect(notifications.length).toBe(exampleNotifications.length);
+        notifications.forEach((notification, index) => {
+          expect(notification.textContent).toEqual(exampleNotifications[index]);
+        });
+      });
+    });
+
+    describe('when no notifications are fired by the pipeline trigger', () => {
+      beforeEach(() => {
+        notificationsState = {
+          notifications: [],
+        };
+      });
+
+      it('should not render the notifications component', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const notifications = element.shadowRoot.querySelectorAll(
+          selectors.notifications
+        );
+
+        notifications.forEach((notification) => {
+          expect(notification.textContent).toEqual(null);
+        });
+        expect(notifications.length).toEqual(0);
+      });
+    });
+
+    it('should subscribe to the headless state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
+    });
```

Comment:
It's better to organize this in a better way and include more tests, 
also it's better to follow a logic order of the test specs, for example this one would be on top of the tests that checks the presence or the absence of the notifications.

```
  describe('component  initialization', () => {
    it('should build the controller with the right parameters', async () => {
      createTestComponent();
      await flushPromises();

      expect(functionsMocks.buildUserActions).toHaveBeenCalledTimes(1);
      expect(functionsMocks.buildUserActions).toHaveBeenCalledWith(
        exampleEngine,
      );

    it('should subscribe to the headless state changes', async () => {
      createTestComponent();
      await flushPromises();

      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
    });
    });
```